invenio_circulation/views.py

```diff
@@ -206,3 +207,81 @@ def get(self, *args, **kwargs):
             )
 
         return jsonify({})
+
+
+def loan_update_blueprint(app):
+    """Create a blueprint for updating Loan."""
+    blueprint = Blueprint(
+        "invenio_circulation_update_loan", __name__, url_prefix=""
+    )
+
+    create_error_handlers(blueprint)
+    rec_serializers = {
+        "application/json": (
+            "invenio_records_rest.serializers" ":json_v1_response"
+        )
+    }
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+    loan_request = LoanUpdateResource.as_view(
+        LoanUpdateResource.view_name.format(CIRCULATION_LOAN_PID_TYPE),
+        serializers=serializers,
+        ctx={},
+    )
+
+    url = "circulation/loans/<{0}:pid_value>/update".format(LOANID_CONVERTER)
+    blueprint.add_url_rule(url, view_func=loan_request, methods=["POST"])
+    return blueprint
+
+
+def validate_update(data):
+    """Validate data before update loan."""
+    item_pid = data.get("item_pid")
+    item_exists = current_app.config.get("CIRCULATION_ITEM_EXISTS")(item_pid)
+    if not item_pid or item_pid == "" or not item_exists:
+        raise BadRequest()
+
+
+def loan_update(loan, data):
+    """Update loan with new properties."""
+    item_ref_builder = current_app.config.get("CIRCULATION_ITEM_REF_BUILDER")
+    item_pid = data.get("item_pid")
+    loan["item_pid"] = item_pid
+    loan["item"] = item_ref_builder(item_pid)
+    return loan
+
+
+class LoanUpdateResource(ContentNegotiatedMethodView):
+    """Loan update resource."""
+
+    view_name = "loan_update_resource"
+
+    def __init__(self, serializers, ctx, *args, **kwargs):
+        """Constructor."""
+        super(LoanUpdateResource, self).__init__(serializers, *args, **kwargs)
+        for key, value in ctx.items():
+            setattr(self, key, value)
+
+    @need_permissions("loan-actions")
+    @pass_record
+    def post(self, pid, record, *args, **kwargs):
+        """Handle POST request to update loan with new item."""
+        old_item_pid = record["item_pid"]
```

Comment:
Should we fail if the item if:
```
record['status'] not in current_app.config['CIRCULATION_STATES_LOAN_ACTIVE']
```
?
I think this will ensure data consistency.